File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int i=0;
        for (int x : nums){
            int a = target - x;
            if (map.containsKey(a)){
                return new int[]{map.get(a),i};
            }
            else{
                map.put(x,i);
            }
            i++;
        }
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> freq;
        for (int i=0; i<nums.size(); i++){
            int needed = target - nums[i];
            if (freq.count(needed)){
                return {freq[needed],i};
                break;
            }
            freq[nums[i]] = i;
        }
        return {};
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna